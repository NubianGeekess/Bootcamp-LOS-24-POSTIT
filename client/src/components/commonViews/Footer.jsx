import React from 'react';

/**
 * Footer Component
 * Displays page footer where required
 *
 * @method Footer
 *
 * @returns {Object} JSX
 */
const Footer = () =>
	<footer className="container page-footer">
		<div className="divider"></div>
		<ul className="right">
			<li>
				<a href="api-docs">
					<i className="fa fa-file-code-o"></i>&nbsp;&nbsp;
					<span>api docs</span>
				</a>
			</li>
			<li>
				<a href="https://github.com/dinobi/Bootcamp-LOS-24-POSTIT">
					<i className="fa fa-github"></i>&nbsp;&nbsp;
					<span>github</span>
				</a>
			</li>
		</ul>
		<div className="copyright left">
			<small>&copy;Bootcamp24, Andela Nigeria. All rights reserved.</small>
		</div>
	</footer>;

export default Footer;
