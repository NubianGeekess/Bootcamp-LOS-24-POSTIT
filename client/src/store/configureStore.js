import { compose, createStore, applyMiddleware } from 'redux';
import reduxImmutableStateInvariant from 'redux-immutable-state-invariant';
import { persistStore, autoRehydrate } from 'redux-persist';
import thunk from 'redux-thunk';
import rootReducer from '../reducers';

// Be sure to ONLY add this middleware in development!
const store = process.env.NODE_ENV === 'development' ?
  createStore(
    rootReducer,
    compose(
      applyMiddleware(...[reduxImmutableStateInvariant(), thunk]),
      window.__REDUX_DEVTOOLS_EXTENSION__ && window.__REDUX_DEVTOOLS_EXTENSION__()
    )
  )
  :
  createStore(
    rootReducer,
    applyMiddleware(thunk),
  );

// persistStore(store);

export default store;
