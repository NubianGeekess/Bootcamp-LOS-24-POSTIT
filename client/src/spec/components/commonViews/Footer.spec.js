/* globals expect */
import React from 'react';
import PropTypes from 'prop-types';
import { Provider } from 'react-redux';
import { MemoryRouter } from 'react-router-dom';
import configureStore from 'redux-mock-store';
import Footer from '../../../components/commonViews/Footer';
import mockData from '../../mocks/mockData';

describe('Given Footer component is mounted', () => {
  it('should render self without crashing', () => {
    const wrapper = shallow(<Footer />);
    expect(wrapper.exists()).toBe(true);
    expect(wrapper.find('footer').exists()).toBe(true);
  });
});