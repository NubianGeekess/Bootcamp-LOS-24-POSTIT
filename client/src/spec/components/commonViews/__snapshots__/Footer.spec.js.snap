// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given Footer component is mounted should render self without crashing 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Footer />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="divider"
        />,
        <ul
          className="right"
        >
          <li>
            <a>
              <i
                className="fa fa-google-plus"
              />
            </a>
          </li>
          <li>
            <a
              href="https://github.com/dinobi/Bootcamp-LOS-24-POSTIT"
            >
              <i
                className="fa fa-github"
              />
            </a>
          </li>
          <li>
            <a>
              <i
                className="fa fa-facebook"
              />
            </a>
          </li>
          <li>
            <a>
              <i
                className="fa fa-twitter"
              />
            </a>
          </li>
        </ul>,
        <div
          className="copyright left"
        >
          <small>
            ©Bootcamp24, Andela Nigeria. All rights reserved.
          </small>
        </div>,
      ],
      "className": "container page-footer",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "divider",
        },
        "ref": null,
        "rendered": null,
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <li>
              <a>
                <i
                  className="fa fa-google-plus"
                />
              </a>
            </li>,
            <li>
              <a
                href="https://github.com/dinobi/Bootcamp-LOS-24-POSTIT"
              >
                <i
                  className="fa fa-github"
                />
              </a>
            </li>,
            <li>
              <a>
                <i
                  className="fa fa-facebook"
                />
              </a>
            </li>,
            <li>
              <a>
                <i
                  className="fa fa-twitter"
                />
              </a>
            </li>,
          ],
          "className": "right",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <a>
                <i
                  className="fa fa-google-plus"
                />
              </a>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <i
                  className="fa fa-google-plus"
                />,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "className": "fa fa-google-plus",
                },
                "ref": null,
                "rendered": null,
                "type": "i",
              },
              "type": "a",
            },
            "type": "li",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <a
                href="https://github.com/dinobi/Bootcamp-LOS-24-POSTIT"
              >
                <i
                  className="fa fa-github"
                />
              </a>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <i
                  className="fa fa-github"
                />,
                "href": "https://github.com/dinobi/Bootcamp-LOS-24-POSTIT",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "className": "fa fa-github",
                },
                "ref": null,
                "rendered": null,
                "type": "i",
              },
              "type": "a",
            },
            "type": "li",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <a>
                <i
                  className="fa fa-facebook"
                />
              </a>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <i
                  className="fa fa-facebook"
                />,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "className": "fa fa-facebook",
                },
                "ref": null,
                "rendered": null,
                "type": "i",
              },
              "type": "a",
            },
            "type": "li",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <a>
                <i
                  className="fa fa-twitter"
                />
              </a>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <i
                  className="fa fa-twitter"
                />,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "className": "fa fa-twitter",
                },
                "ref": null,
                "rendered": null,
                "type": "i",
              },
              "type": "a",
            },
            "type": "li",
          },
        ],
        "type": "ul",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <small>
            ©Bootcamp24, Andela Nigeria. All rights reserved.
          </small>,
          "className": "copyright left",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "©Bootcamp24, Andela Nigeria. All rights reserved.",
          },
          "ref": null,
          "rendered": "©Bootcamp24, Andela Nigeria. All rights reserved.",
          "type": "small",
        },
        "type": "div",
      },
    ],
    "type": "footer",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="divider"
          />,
          <ul
            className="right"
          >
            <li>
              <a>
                <i
                  className="fa fa-google-plus"
                />
              </a>
            </li>
            <li>
              <a
                href="https://github.com/dinobi/Bootcamp-LOS-24-POSTIT"
              >
                <i
                  className="fa fa-github"
                />
              </a>
            </li>
            <li>
              <a>
                <i
                  className="fa fa-facebook"
                />
              </a>
            </li>
            <li>
              <a>
                <i
                  className="fa fa-twitter"
                />
              </a>
            </li>
          </ul>,
          <div
            className="copyright left"
          >
            <small>
              ©Bootcamp24, Andela Nigeria. All rights reserved.
            </small>
          </div>,
        ],
        "className": "container page-footer",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "divider",
          },
          "ref": null,
          "rendered": null,
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <li>
                <a>
                  <i
                    className="fa fa-google-plus"
                  />
                </a>
              </li>,
              <li>
                <a
                  href="https://github.com/dinobi/Bootcamp-LOS-24-POSTIT"
                >
                  <i
                    className="fa fa-github"
                  />
                </a>
              </li>,
              <li>
                <a>
                  <i
                    className="fa fa-facebook"
                  />
                </a>
              </li>,
              <li>
                <a>
                  <i
                    className="fa fa-twitter"
                  />
                </a>
              </li>,
            ],
            "className": "right",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <a>
                  <i
                    className="fa fa-google-plus"
                  />
                </a>,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <i
                    className="fa fa-google-plus"
                  />,
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "className": "fa fa-google-plus",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "i",
                },
                "type": "a",
              },
              "type": "li",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <a
                  href="https://github.com/dinobi/Bootcamp-LOS-24-POSTIT"
                >
                  <i
                    className="fa fa-github"
                  />
                </a>,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <i
                    className="fa fa-github"
                  />,
                  "href": "https://github.com/dinobi/Bootcamp-LOS-24-POSTIT",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "className": "fa fa-github",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "i",
                },
                "type": "a",
              },
              "type": "li",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <a>
                  <i
                    className="fa fa-facebook"
                  />
                </a>,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <i
                    className="fa fa-facebook"
                  />,
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "className": "fa fa-facebook",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "i",
                },
                "type": "a",
              },
              "type": "li",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <a>
                  <i
                    className="fa fa-twitter"
                  />
                </a>,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <i
                    className="fa fa-twitter"
                  />,
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "className": "fa fa-twitter",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "i",
                },
                "type": "a",
              },
              "type": "li",
            },
          ],
          "type": "ul",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <small>
              ©Bootcamp24, Andela Nigeria. All rights reserved.
            </small>,
            "className": "copyright left",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "©Bootcamp24, Andela Nigeria. All rights reserved.",
            },
            "ref": null,
            "rendered": "©Bootcamp24, Andela Nigeria. All rights reserved.",
            "type": "small",
          },
          "type": "div",
        },
      ],
      "type": "footer",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactFifteenAdapter {
      "options": Object {
        "supportPrevContextArgumentOfComponentDidUpdate": true,
      },
    },
  },
}
`;
